### PR TITLE
Everywhere: Convert `vm.call` to `JS::call` and remove it from vm

### PR DESCRIPTION
**Pull request by mjz19910**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/VM.cpp [hunk 1]
```diff
@@ -543,16 +543,6 @@ Value VM::get_new_target()
     return verify_cast<FunctionEnvironment>(env).new_target();
 }
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/Runtime/VM.cpp:545]: This commit's title should say "Remove VM::call()`, not `JS::call()`.

**Comment by linusg** [comment]
Looks good, thanks. I don't understand why you add an FunctionObject overload and then still call the Value one in most cases? Given that all of these instances have a FunctionObject, they should also call that overload *except* where we do a type check at the call site, that could be done by the AO.
I'd also suggest extending the first commit to add a call_impl that receives a FunctionObject and doesn't have to do the type check.

**Comment by mjz19910** [comment]
Ok, now the commit messages are a bit better, before the JS::call overload that was added had nothing in common with the commit message (added a JS::call overload for Optional<MarkedValueList>). Now avoids conversion into value and back to function when possible

File: Userland/Libraries/LibJS/Runtime/PromiseReaction.h [hunk 2]
```diff
@@ -20,22 +20,22 @@ struct PromiseCapability {
...
+#define TRY_OR_REJECT(vm, global_object, capability, expression)                                                                        \
+    ({                                                                                                                                  \
+        auto _temporary_try_or_reject_result = (expression);                                                                            \
+        /* 1. If value is an abrupt completion, then */                                                                                 \
+        if (_temporary_try_or_reject_result.is_error()) {                                                                               \
+            vm.clear_exception();                                                                                                       \
+                                                                                                                                        \
+            /* a. Perform ? Call(capability.[[Reject]], undefined, « value.[[Value]] »). */                                           \
```
Comment (mjz19910) [inline Userland/Libraries/LibJS/Runtime/PromiseReaction.h:30]: clang-format does not care that these unicode chars display shorter, just assumes each byte is the same size...

**Review by linusg — COMMENTED** [review]
We don't want to include the JS namespace prefix everywhere though, just outside of LibJS or where call conflicts with FunctionObject::call.
(Easiest way to fix is probably search & replace to remove it everywhere and then add it back where the compiler complains)

File: Userland/Libraries/LibJS/Runtime/PromiseReaction.h [hunk 3]
```diff
@@ -20,22 +20,22 @@ struct PromiseCapability {
...
-            TRY(vm.call(*capability.reject, js_undefined(), *_temporary_try_or_reject_result.release_error().value())); \
-                                                                                                                        \
-            /* b. Return capability.[[Promise]]. */                                                                     \
-            return capability.promise;                                                                                  \
-        }                                                                                                               \
-                                                                                                                        \
-        /* 2. Else if value is a Completion Record, set value to value.[[Value]]. */                                    \
-        _temporary_try_or_reject_result.release_value();                                                                \
+#define TRY_OR_REJECT(vm, global_object, capability, expression)                                                                        \
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/PromiseReaction.h:23]: Please only pass a global object if you need both, and get the VM from that.

**Review by linusg — APPROVED** [review]
Perfect, thanks again!

**Pull request merged**